#! /bin/sh /usr/share/dpatch/dpatch-run
## 99_fix_parse-datetime-test.dpatch by  <93sam@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Apply fix from upstream so that test-parse-datetime will work
## DP: regardless of DST.
## DP: http://thread.gmane.org/gmane.comp.lib.gnulib.bugs/28825

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' coreutils-8.13~/gnulib-tests/test-parse-datetime.c coreutils-8.13/gnulib-tests/test-parse-datetime.c
--- coreutils-8.13~/gnulib-tests/test-parse-datetime.c	2011-08-31 18:01:21.000000000 +0100
+++ coreutils-8.13/gnulib-tests/test-parse-datetime.c	2013-01-26 14:04:38.000000000 +0000
@@ -96,20 +96,17 @@
 #endif /* ! HAVE_TM_GMTOFF */
 
 static long
-gmt_offset ()
+gmt_offset (time_t s)
 {
-  time_t now;
   long gmtoff;
 
-  time (&now);
-
 #if !HAVE_TM_GMTOFF
-  struct tm tm_local = *localtime (&now);
-  struct tm tm_gmt   = *gmtime (&now);
+  struct tm tm_local = *localtime (&s);
+  struct tm tm_gmt   = *gmtime (&s);
 
   gmtoff = tm_diff (&tm_local, &tm_gmt);
 #else
-  gmtoff = localtime (&now)->tm_gmtoff;
+  gmtoff = localtime (&s)->tm_gmtoff;
 #endif
 
   return gmtoff;
@@ -125,16 +122,17 @@
   const char *p;
   int i;
   long gmtoff;
+  time_t ref_time = 1304250918;
 
   set_program_name (argv[0]);
 
-  gmtoff = gmt_offset ();
+  gmtoff = gmt_offset (ref_time);
 
 
   /* ISO 8601 extended date and time of day representation,
      'T' separator, local time zone */
   p = "2011-05-01T11:55:18";
-  expected.tv_sec = 1304250918 - gmtoff;
+  expected.tv_sec = ref_time - gmtoff;
   expected.tv_nsec = 0;
   ASSERT (parse_datetime (&result, p, 0));
   LOG (p, expected, result);
@@ -144,7 +142,7 @@
   /* ISO 8601 extended date and time of day representation,
      ' ' separator, local time zone */
   p = "2011-05-01 11:55:18";
-  expected.tv_sec = 1304250918 - gmtoff;
+  expected.tv_sec = ref_time - gmtoff;
   expected.tv_nsec = 0;
   ASSERT (parse_datetime (&result, p, 0));
   LOG (p, expected, result);
@@ -155,7 +153,7 @@
   /* ISO 8601, extended date and time of day representation,
      'T' separator, UTC */
   p = "2011-05-01T11:55:18Z";
-  expected.tv_sec = 1304250918;
+  expected.tv_sec = ref_time;
   expected.tv_nsec = 0;
   ASSERT (parse_datetime (&result, p, 0));
   LOG (p, expected, result);
